text/0000-contract-metadata.md

```diff
@@ -0,0 +1,117 @@
+- Proposal Name: `contract_metadata`
```

Comment:
Consider renaming to contract ABI -- this is the common term used among blockchains.

Reply:
it's not just ABI though, we also need UX-specific info (like contract name, icon, etc).